commit 4642f4eead07080bf46b6d67b873d303eda9fb48
Author: Khem Raj <raj.khem@gmail.com>
Date:   Sat Jun 11 00:15:02 2011 -0700

    Rules.mak: Rearrange appending UCLIBC_EXTRA_CFLAGS to CFLAGS
    
    UCLIBC_EXTRA_CFLAGS is currently added before the OPTIMIZATION flags
    and OPTIMIZATION is chosen to be Os by default. But in OE we pass the optimisation
    flags through UCLIBC_EXTRA_CFLAGS but they are not effective since -Os is
    specified at last. So we need to change the order of these option flags
    
    Signed-off-by: Khem Raj <raj.khem@gmail.com>
    Signed-off-by: Bernhard Reutner-Fischer <rep.dot.nop@gmail.com>

diff --git a/Rules.mak b/Rules.mak
index a00deea..6b8f5b2 100644
--- a/Rules.mak
+++ b/Rules.mak
@@ -585,9 +585,6 @@ CFLAGS := -include $(top_srcdir)include/libc-symbols.h \
 	-nostdinc -I$(top_builddir)include -I$(top_srcdir)include -I. \
 	-I$(top_srcdir)libc/sysdeps/linux \
 	-I$(top_srcdir)libc/sysdeps/linux/$(TARGET_ARCH)
-ifneq ($(strip $(UCLIBC_EXTRA_CFLAGS)),"")
-CFLAGS += $(call qstrip,$(UCLIBC_EXTRA_CFLAGS))
-endif
 
 # We need this to be checked within libc-symbols.h
 ifneq ($(HAVE_SHARED),y)
@@ -633,6 +630,9 @@ LDFLAGS += -Wl,-s
 else
 STRIPTOOL := true -Stripping_disabled
 endif
+ifneq ($(strip $(UCLIBC_EXTRA_CFLAGS)),"")
+CFLAGS += $(call qstrip,$(UCLIBC_EXTRA_CFLAGS))
+endif
 
 ifeq ($(DOMULTI),y)
 # we try to compile all sources at once into an object (IMA), but
